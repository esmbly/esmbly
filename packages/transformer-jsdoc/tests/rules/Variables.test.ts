import testRunner from '../helpers/stringTestRunner';

describe('rule: Variables', () => {
  it('transforms a variable declaration to TypeScript (var)', async () => {
    const program = `
      /**
       * @type {number}
       */
      var a = 1;
    `;
    const [{ content }] = await testRunner(program);
    expect(content).toMatchSnapshot();
  });

  it('transforms a variable declaration to TypeScript (let)', async () => {
    const program = `
      /**
       * @type {number}
       */
      let b = 1;
    `;
    const [{ content }] = await testRunner(program);
    expect(content).toMatchSnapshot();
  });

  it('transforms a variable declaration to TypeScript (const)', async () => {
    const program = `
      /**
       * @type {number}
       * @const
       */
      const c = 2;
    `;
    const [{ content }] = await testRunner(program);
    expect(content).toMatchSnapshot();
  });

  it('transforms a variable declaration to TypeScript (exported)', async () => {
    const program = `
      /**
       * @type {number}
       * @const
       */
      export const c = 2;
    `;
    const [{ content }] = await testRunner(program);
    expect(content).toMatchSnapshot();
  });
});
