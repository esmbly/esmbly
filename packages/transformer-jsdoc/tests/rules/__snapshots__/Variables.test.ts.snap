// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rule: Variables @const is a alias for @constant 1`] = `
"
        /**
         * @const {number}
         */
        const a: number = 1;
      "
`;

exports[`rule: Variables @constant transforms a variable declaration to TypeScript (const) 1`] = `
"
        /**
         * @constant {number}
         */
        const c: number = 2;
      "
`;

exports[`rule: Variables @constant transforms a variable declaration to TypeScript (let) 1`] = `
"
        /**
         * @constant {number}
         */
        const b: number = 1;
      "
`;

exports[`rule: Variables @constant transforms a variable declaration to TypeScript (var) 1`] = `
"
        /**
         * @constant {number}
         */
        const a: number = 1;
      "
`;

exports[`rule: Variables @declare adds the declare keyword (const) 1`] = `
"
        /**
         * @type {number}
         * @declare
         */
        declare const a: number = 1;
      "
`;

exports[`rule: Variables @declare adds the declare keyword (let) 1`] = `
"
        /**
         * @type {number}
         * @declare
         */
        declare let a: number;
      "
`;

exports[`rule: Variables @declare adds the declare keyword (var) 1`] = `
"
        /**
         * @type {number}
         * @declare
         */
        declare var a: number;
      "
`;

exports[`rule: Variables @type transforms a variable declaration to TypeScript (const) 1`] = `
"
        /**
         * @type {number}
         * @const
         */
        const c: number = 2;
      "
`;

exports[`rule: Variables @type transforms a variable declaration to TypeScript (exported const) 1`] = `
"
        /**
         * @type {number}
         * @const
         */
        export const c: number = 2;
      "
`;

exports[`rule: Variables @type transforms a variable declaration to TypeScript (let) 1`] = `
"
        /**
         * @type {number}
         */
        let b: number = 1;
      "
`;

exports[`rule: Variables @type transforms a variable declaration to TypeScript (var) 1`] = `
"
        /**
         * @type {number}
         */
        var a: number = 1;
      "
`;
